### DISCUSSION TITLE
Terminate running script using Dagster Shell

### DISCUSSION THREAD

**Discussion by cody-scott** [dagster-io/dagster#19053] [Q&A] [answered]
Hi,

Curious if someone can help me understand how to terminate a running script executed by `execute_shell_command`. In my case I terminated the job (from the UI), but the underlying script continued to process.

Is there a pattern or way to terminate the underlying process if the run gets terminated?

Thanks!

**Comment by alangenfeld** [comment, marked as answer]
We should be sending a `SIGTERM` to the underlying process already https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-shell/dagster_shell/utils.py#L129-L132

Do you have more details on the platform and what type of process was running that didn't terminate?

**Reply by cody-scott** [reply]
Sorry yes, it was running an r script triggered by 

```python
script_args = ["Rscript", "path/to/script.r"]
msg, status_code = execute_shell_command(
            " ".join(script_args),
            "STREAM",
            context.log,
        )
```

We were noticing the db connection was still running in the background, even though it had terminated. Might be more of an R issue rather then dagster, but open to suggestions.

**Reply by alangenfeld** [reply]
I would look in to setting up graceful shutdown on `SIGTERM` in your R program, a quick search brings up https://github.com/atheriel/sigterm for example

**Reply by cody-scott** [reply]
Thanks!
